libcodechecker/cmd/cmd_line_client.py

```diff
@@ -259,54 +262,198 @@ def handle_list_results(args):
 
 
 def handle_diff_results(args):
-    def printResult(getterFn, baseid, newid, suppr, output_format):
+    def getDiffResults(getterFn, baseid, newid, suppr):
         report_filter = [
             codeCheckerDBAccess.ttypes.ReportFilter(suppressed=suppr)]
         add_filter_conditions(report_filter[0], args.filter)
-
-        sort_type = None
+        sort_mode = []
+        sort_mode.append(codeCheckerDBAccess.ttypes.SortMode(
+            codeCheckerDBAccess.ttypes.SortType.FILENAME,
+            codeCheckerDBAccess.ttypes.Order.ASC))
         limit = codeCheckerDBAccess.constants.MAX_QUERY_SIZE
         offset = 0
 
         all_results = []
-        results = getterFn(baseid, newid, limit, offset, sort_type,
+        results = getterFn(baseid, newid, limit, offset, sort_mode,
                            report_filter)
 
         while results:
             all_results.extend(results)
             offset += limit
-            results = getterFn(baseid, newid, limit, offset, sort_type,
+            results = getterFn(baseid, newid, limit, offset, sort_mode,
                                report_filter)
+        return all_results
+
+    def getReportDirResults(reportdir):
+        all_reports = []
+        for filename in os.listdir(reportdir):
+            if filename.endswith(".plist"):
+                file_path = os.path.join(reportdir, filename)
+                LOG.debug("Parsing:" + file_path)
+                try:
+                    files, reports = plist_parser.parse_plist(file_path)
+                    for report in reports:
+                        report.main['location']['file_name'] = \
+                            files[int(report.main['location']['file'])]
+                    all_reports.extend(reports)
+
+                except Exception as ex:
+                    LOG.error('The generated plist is not valid!')
+                    LOG.error(ex)
+        return all_reports
+
+    def getLineFromFile(filename, lineno):
+        with open(filename, 'r') as f:
+            i = 1
+            for line in f:
+                if i == lineno:
+                    return line
+                i += 1
+        return ""
+
+    def getLineFromRemoteFile(client, fid, lineno):
+        # FIXME: large special file content
```

Comment:
Maybe we should mention that "special" file content in this case is the UTF-8 encoding problem?